### DISCUSSION TITLE
Minimum memory needed?

### DISCUSSION THREAD

**Discussion by rygo6** [wasmerio/wasmer#3697] [General]
I am trying to understand how much memory wasmer requires to run total. Not how much memory just the wasm code needs, but how much memory the entire runtime needs for the wasm code itself, for any guard pages, or anything else required.

For instance, wasmtime says that all linear memory regions are preceded by a 2gb guard region: https://docs.wasmtime.dev/security.html

Is this also true for wasmer? Do you need at least a 2gb guard region for any wasm vm instance?

I also see that wasmer has some settings to change guard page sizes:
https://docs.rs/wasmer/1.0.0-alpha01.0/wasmer/struct.Tunables.html
Is there some rule for the minimum size a guard page must be in order to appropriately secure the VM?

**Comment by syrusakbary** [comment]
By default Wasmer allocates 4Gb+2Gb memory guard virtual memory in case the WebAssembly memory doesn't define a maximum, to make sure the compiled Wasm modules run at max speed, although it will be custom depending on the system.
Wasmer also do a default 2Gb guard region for Wasm instances.

You can customize this behavior via Wasmer Tunables https://docs.rs/wasmer/3.1.0/wasmer/trait.Tunables.html
